components/driver/uart.c

```diff
@@ -1281,12 +1283,15 @@ esp_err_t uart_flush_input(uart_port_t uart_num)
         }
         data = (uint8_t*) xRingbufferReceive(p_uart->rx_ring_buf, &size, (portTickType) 0);
         if(data == NULL) {
+            UART_ENTER_CRITICAL(&(uart_context[uart_num].spinlock));
             if( p_uart_obj[uart_num]->rx_buffered_len != 0 ) {
-                ESP_LOGE(UART_TAG, "rx_buffered_len error");
                 p_uart_obj[uart_num]->rx_buffered_len = 0;
+                UART_EXIT_CRITICAL(&(uart_context[uart_num].spinlock));
+                // this must be called outside the critical section
+                ESP_LOGE(UART_TAG, "rx_buffered_len error");
```

Comment:
It would be better to set a flag in this case and print this log message after `UART_EXIT_CRITICAL` and before `break;` in case the flag is set. Entering and exiting critical sections is rather expensive compared to setting and testing a flag

Reply:
Why do you need to care about performance in an unlikely error path?

Reply:
You are right about performance, in fact, I meant that we would prefer to have both variable set atomically, in order to avoid being pre-empted in-between

Reply:
> ), whereas `uart_get_stop_bits` and `uart_get_parity` are atomic.

You mean this is atomic?
https://github.com/espressif/esp-idf/blob/master/components/hal/esp32/include/hal/uart_ll.h#L381
https://github.com/espressif/esp-idf/blob/master/components/hal/esp32/include/hal/uart_ll.h#L347